app/post.go

```diff
@@ -1331,6 +1331,38 @@ func (a *App) MaxPostSize() int {
 	return a.Srv().MaxPostSize()
 }
 
+// countThreadMentions returns the number of times the user is mentioned in a specified thread after the timestamp.
+func (a *App) countThreadMentions(user *model.User, post *model.Post, timestamp int64) (int64, *model.AppError) {
```

Comment:
What means mention?, that is important, it means explicit mentions, or also for example means, DM message?

Reply:
explicit mentions only. I think

Reply:
I think this should handle any kind of thing that is consider a mention, based on the user configuration. Basically we need to follow exactly the same code that is used to update the mention count. For example, in DMs are mentions, the add-to-channel is a mention if the post is for the your user, group mentions... basically yo need to replicate what is done in the `notifications.go` code unless the unread of the channel was behaving in a different way.

Reply:
@jespino - I don't think I agree. Neither DM mentions nor group mentions are connected with threads.
We add autofollow feature to 'bring people into the conversation'. So I think only explicit mentions should trigger the autofollow. Let's see what @esethna thinks about this :)

Reply:
This is a really good question. Please let's ask @matthewbirtch. ie Matt the question is about thread replies in DMs or GMs that don't have an explicit @mention in them, would we be treating them with a mention badge or not (ie currently all DM messages are treated as mentions)?

Reply:
I'm not sure I full understand the question, but here are my thoughts based on how I'm interpreting the conversation. Let me know if I'm interpreting incorrectly:

**autofollowing**
- threads in DMs should be auto followed since you are implicitly participating and getting mentioned, but it's only a "thread" once a reply has occurred 
- threads in GMs should only be auto followed if you are mentioned or if you have have replied to that thread. Also, it's only a "thread" once a reply has occurred

**mention badges**
- for DMs, show mention badge on the DM sidebar item if it's a single parent post
- if it's a reply to a thread in a DM, show the mention badge on the Threads sidebar item rather than the DM sidebar item
- for GMs, show mention badge on the GM sidebar item if it's a single parent post that you've been mentioned in
- if you are mentioned in a reply to a thread in a GM, that thread is then auto followed and show the badge on the Threads sidebar item rather than the GM sidebar item

Reply:
@matthewbirtch what about `@group` mentions? Should all members of that group autofollow the thread in which that mention was posted? I fear that it might become a little spammy

Reply:
> @matthewbirtch what about `@group` mentions? Should all members of that group autofollow the thread in which that mention was posted? I fear that it might become a little spammy

Good question. A `@group` mention should behave the same as a regular mention, no? I would assume that it should be auto-followed once it has a reply and becomes a true thread.

I'm open to other ideas though

Reply:
@matthewbirtch - I am just worried of this scenario:
User posts a message.
Another user replies and does `@group`.
ALL users in that group will start following that thread. Seemed a little bit 'spammy' to me.

Reply:
> @matthewbirtch - I am just worried of this scenario:
> User posts a message.
> Another user replies and does `@group`.
> ALL users in that group will start following that thread. Seemed a little bit 'spammy' to me.

Ya, I get that, but if we don't do this, then some of the point of a `@group` mention gets lost. I guess it depends on how big some of these groups are. Maybe we have a threshold for how this behaves depending on the size of the group? Not sure if that's possible

Reply:
ok. let's settle on implementing this as-is initially and discuss possibly setting the limits later on

Reply:
Sounds good to me. We'll keep a close eye on this